components/encryption/src/master_key/kms.rs

```diff
@@ -228,20 +227,41 @@ impl State {
         Ok(())
     }
 
-    fn cached(&self, ciphertext_key: &Encrypted) -> bool {
+    fn cached(&self, ciphertext_key: &EncryptedKey) -> bool {
         *ciphertext_key == self.cached_ciphertext_key
     }
 }
 
+// EncryptedKey is a newtype used to mark data as an encrypted key
+// It requires the vec to be non-empty
+#[derive(PartialEq, Clone, Debug)]
+pub struct EncryptedKey(Vec<u8>);
```

Comment:
maybe we don't even want the encrypted key to be printed?

Reply:
In theory it is safe to publish encrypted data publicly, so I am not worried about the security. I think we need some way to identify the data key when debugging.

Reply:
okay